e2e/testcases/oci_sync_test.go

```diff
@@ -475,6 +478,14 @@ func TestSwitchFromGitToOci(t *testing.T) {
 	if err := nt.ValidateNotFound("bookinfo-sa", namespace, &corev1.ServiceAccount{}); err != nil {
 		nt.T.Error(err)
 	}
+	nt.T.Cleanup(func() {
```

Comment:
Do we really want to reset to OCI and then reset again to Git? 
The git cleanup should be sufficient, right?

Reply:
This fixes one of the issues with the field manager swapping, where the `oci` spec is leftover if we go directly to `git` - because the `oci` spec was applied from the test suite field manager.

e.g.
```yaml
  spec:
    git:
      auth: none
      branch: main
      period: 0s
      repo: https://github.com/config-sync-examples/namespace-repo-bookinfo
    oci:
      auth: none
      image: us-docker.pkg.dev/stolos-dev/config-sync-test-public/namespace-repo-bookinfo
    sourceType: oci
```

This was causing an issue because the `repoSyncGit` wasn't specifying a `sourceType`, although that issue should be fixed now with the updated `repoSyncGit`. I don't think we strictly _need_ this block but I think it's good form to clean things up in the reverse order which they were applied in.

Another alternative is that this block could delete the suite-created `RepoSync`.

Reply:
I guess this is fine. It's just slower because it has to wait for sync twice.